### PR TITLE
Move third_party to recirq/third_party

### PR DESCRIPTION
**Pull request by fdmalone**
Should fix pip installation within notebooks.

### CODE REVIEW

File: recirq/third_party/quaff/basis_change_test.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import cirq
 import numpy as np
 import pytest
-import quaff
+from recirq.third_party import quaff as quaff
```
Comment (pavoljuhas) [inline recirq/third_party/quaff/basis_change_test.py:4]: nit - `as quaff` is redundant -
```suggestion
from recirq.third_party import quaff
```
also please fix the same import in comb_test.py, cz_layer_test.py, indexing_test.py, linalg_test.py, random_test.py

File: recirq/third_party/quaff/json_resolver.py [hunk 2]
```diff
@@ -4,9 +4,9 @@
 
 
 def json_resolver(quaff_type: str) -> Optional[ObjectFactory]:
-    if quaff_type.startswith("quaff."):
+    if quaff_type.startswith("quaff"):
         return cast(
             ObjectFactory,
-            eval(f"__import__('quaff').{quaff_type.split('.', 1)[1]}", {}, {}),
+            eval(f"__import__('recirq').third_party.{quaff_type}", {}, {}),
```
Comment (pavoljuhas) [inline recirq/third_party/quaff/json_resolver.py:10]: Please rewrite without eval, for example, after adding module-level import `from recirq.third_party import quaff` -
```suggestion
            getattr(quaff, quaff_type[6:]
```

File: recirq/third_party/quaff/json_resolver.py [hunk 3]
```diff
@@ -4,9 +4,9 @@
 
 
 def json_resolver(quaff_type: str) -> Optional[ObjectFactory]:
-    if quaff_type.startswith("quaff."):
+    if quaff_type.startswith("quaff"):
```
Comment (pavoljuhas) [inline recirq/third_party/quaff/json_resolver.py:7]: The previous prefix `quaff.` seems more robust.
Is there some edge case that does not work with `quaff.` and needs `quaff` instead?

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
Few small suggestions, but otherwise LGTM.

**Comment by pavoljuhas** [comment]
One more comment - after merging with master the high_level.ipynb notebook fails when trying to read `.../site-packages/recirq/qcqmc/data/integrals/fh_sto3g/hamiltonian.chk`.

Perhaps that file needs to be added to package_data here?
https://github.com/quantumlib/ReCirq/blob/17a5fdef73e0e578eb8df33b2c9f8418630ed88c/setup.py#L56-L61

File: recirq/third_party/quaff/basis_change_test.py (hunk 1 above)
Comment (fdmalone) [reply recirq/third_party/quaff/basis_change_test.py:4]: Whoops, thanks.

File: recirq/third_party/quaff/json_resolver.py (hunk 2 above)
Comment (fdmalone) [reply recirq/third_party/quaff/json_resolver.py:10]: Thanks, done.

File: recirq/third_party/quaff/json_resolver.py (hunk 3 above)
Comment (fdmalone) [reply recirq/third_party/quaff/json_resolver.py:7]: No good reason, reverted.

**Comment by fdmalone** [comment]
@pavoljuhas PTAL I think I resolved everything and verified the notebooks run in a clean environment.

**Review by pavoljuhas — APPROVED** [review]

**Pull request merged**